src/test/java/games/strategy/util/MatchTest.java

```diff
@@ -1,26 +0,0 @@
-package games.strategy.util;
-
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertTrue;
-
-import java.util.function.Predicate;
-
-import org.junit.jupiter.api.Test;
-
-public class MatchTest {
```

Comment:
Is there any test coverage on `Predicate`? There was not very much coverage for Match, but if we are removing Match to use Predicate directly, it would seem in scope to update the Match tests to instead check Predicate and to fix test coverage issues.

Reply:
Predicate is part of the java standard library, if it's broken there would be no way to fix it unless copying the class using that instead.